### ISSUE TITLE
How to debug the installation package flow?

### ISSUE THREAD

**Issue by oshliaer** [srcbookdev/srcbook#205]
OK

I want to install the package for web-scraping-with-puppeteer. It seems my internet is slow, and something went wrong. And the installation process falls into a loop or is broken.

How can I debug the installation process?

Cheers!

**Comment by benjreinhart** [comment]
I think we need to fix this properly by rendering the output from npm install, which currently does not happen in the normal npm install flow. Let me try to squeeze in a commit for this shortly!

**Comment by benjreinhart** [comment]
As a stop-gap, we do actually send stdout / stderr over websockets to the client for the package.json install. If you take a look at the websocket messages `cell:output`, you should see the console output.

**Comment by oshliaer** [comment]
@benjreinhart 

I got this again

![2024-08-23_21-25-13](https://github.com/user-attachments/assets/ab09ffd0-d427-4f7e-bf10-1f93fe1017bf)

Great, I see the error if it completed somehow.

![2024-08-23_21-28-15](https://github.com/user-attachments/assets/db6ea674-04e2-4a24-b2e4-26844a89f1cf)

But while the installation is in progress, no socket update occurs.

This is not the application problem, but a problem with my network speed. I tried at a higher speed and could not get this error.

This is probably an interface problem, as you suggested above.

I think the topic can be closed. At this point, it is clear how to view the status.

Thanks!
